docker-entrypoint.sh

```diff
@@ -82,9 +83,15 @@ if [ "$1" = 'rabbitmq-server' ]; then
 			var="RABBITMQ_${conf^^}"
 			val="${!var}"
 			[ "$val" ] || continue
-			cat >> /etc/rabbitmq/rabbitmq.config <<-EOC
-			      {$conf, <<"$val">>},
-			EOC
+			if [[ $conf == default_pass || $conf == default_user || $conf == default_vhost ]]; then
+				cat >> /etc/rabbitmq/rabbitmq.config <<-EOC
+				      {$conf, <<"$val">>},
+				EOC
+			else
+				cat >> /etc/rabbitmq/rabbitmq.config <<-EOC
+				      {$conf, $val},
+				EOC
+			fi
```

Comment:
Sorry for the delay!  I'm +1 on the HiPE change in general (making it available conditionally), but I think this block could use some improvement, and here's what I'd propose instead:
```bash
for conf in "${configs[@]}"; do
	var="RABBITMQ_${conf^^}"
	val="${!var}"
	rawVal=
	case "$conf" in
		# SSL-related options are configured above, so should be ignored here
		ssl_*) continue ;;

		# convert shell booleans into Erlang booleans
		hipe_compile)
			[ "$val" ] && rawVal='true' || rawVal='false'
			;;

		# otherwise, assume string-based (and skip or add appropriate decorations)
		*)
			[ "$val" ] || continue
			rawVal='<<"'"$val"'">>'
			;;
	esac
	cat >> /etc/rabbitmq/rabbitmq.config <<-EOC
		{$conf, $rawVal},
	EOC
done
```

(I'm happy to take over the PR from here as well, if you'd rather, taking your commits and making a new PR that implements this `case`. :+1:)